bin/python/ripple/util/Sign.py

```diff
@@ -44,36 +42,76 @@ def make_seed(urandom=os.urandom):
     return urandom(16)
 
 def make_ed25519_keypair(urandom=os.urandom):
-    private_key = urandom(32)
-    return private_key, ed25519.publickey(private_key)
+    sk = urandom(32)
+    return sk, ed25519.publickey(sk)
 
-def make_ecdsa_keypair():
+def make_ecdsa_keypair(urandom=None):
     # This is not used.
-    private_key = ecdsa.SigningKey.generate(curve=ecdsa.SECP256k1)
+    sk = ecdsa.SigningKey.generate(curve=ecdsa.SECP256k1, entropy=urandom)
     # Can't be unit tested easily - need a mock for ecdsa.
-    vk = private_key.get_verifying_key()
-    sig = private_key.sign('message')
+    vk = sk.get_verifying_key()
+    sig = sk.sign('message')
     assert vk.verify(sig, 'message')
-    return private_key, vk
+    return sk, vk
 
 def make_seed_from_passphrase(passphrase):
     # For convenience, like say testing against rippled we can hash a passphrase
     # to get the seed. validation_create (Josh may have killed it by now) takes
     # an optional arg, which can be a base58 encoded seed, or a passphrase.
     return hashlib.sha512(passphrase).digest()[:16]
 
-def make_manifest(public_key, validator_public_key, seq):
+def make_manifest(master_pk, validation_pk, seq):
+    """create a manifest
+
+    Parameters
+    ----------
+    master_pk : string
+        validator's master public key (_not_ BASE58 encoded)
+    validation_pk : string
+        validator's validation public key (_not_ BASE58 encoded)
+    seq : int
+        manifest sequence number
+
+    Returns
+    ----------
+    string
+        String with fields for seq, master_pk, validation_pk
+    """
     return ''.join([
         SField.sfSequence,
         to_int32(seq),
-        SField.sfPublicKey,      # Master public key.
-        prepend_length_byte(public_key),
-        SField.sfSigningPubKey,  # Ephemeral public key.
-        prepend_length_byte(validator_public_key)])
-
-def sign_manifest(manifest, private_key, public_key):
-    sig = ed25519.signature('MAN\0' + manifest, private_key, public_key)
-    return manifest + SField.sfSignature + prepend_length_byte(sig)
+        SField.sfPublicKey,
+        prepend_length_byte(master_pk),
+        SField.sfSigningPubKey,
+        prepend_length_byte(validation_pk)])
+
+def sign_manifest(manifest, validation_sk, master_sk, master_pk):
```

Comment:
Those are good ideas, however I'm not having success getting either to work.
I think this is the missing Python functionality that we'd need:
https://github.com/ripple/rippled/blob/develop/src/ripple/crypto/impl/GenerateDeterministicKey.cpp#L119